cpp/cmake_modules/FindgRPCAlt.cmake

```diff
@@ -24,222 +24,57 @@ if(gRPC_FOUND)
-if(GRPC_ALT_VERSION)
-  list(APPEND GRPC_ALT_FIND_PACKAGE_ARGS VERSION_VAR GRPC_ALT_VERSION)
-endif()
-find_package_handle_standard_args(${GRPC_ALT_FIND_PACKAGE_ARGS})
-
 if(gRPCAlt_FOUND)
-  add_library(gRPC::gpr UNKNOWN IMPORTED)
-  set_target_properties(gRPC::gpr
-                        PROPERTIES IMPORTED_LOCATION "${GRPC_GPR_LIB}"
-                                   INTERFACE_INCLUDE_DIRECTORIES "${GRPC_INCLUDE_DIR}")
-
-  add_library(gRPC::grpc UNKNOWN IMPORTED)
-  set_target_properties(
-    gRPC::grpc
-    PROPERTIES IMPORTED_LOCATION
-               "${GRPC_GRPC_LIB}"
-               INTERFACE_INCLUDE_DIRECTORIES
-               "${GRPC_INCLUDE_DIR}"
-               INTERFACE_LINK_LIBRARIES
-               "OpenSSL::SSL;OpenSSL::Crypto;ZLIB::ZLIB;c-ares::cares")
-
-  set(_GRPCPP_LINK_LIBRARIES "gRPC::grpc;gRPC::gpr")
-
-  if(GRPC_ADDRESS_SORTING_LIB)
-    # Address sorting is optional and not always required.
-    add_library(gRPC::address_sorting UNKNOWN IMPORTED)
-    set_target_properties(gRPC::address_sorting
-                          PROPERTIES IMPORTED_LOCATION "${GRPC_ADDRESS_SORTING_LIB}"
-                                     INTERFACE_INCLUDE_DIRECTORIES "${GRPC_INCLUDE_DIR}")
-    set(_GRPCPP_LINK_LIBRARIES "${_GRPCPP_LINK_LIBRARIES};gRPC::address_sorting")
-  endif()
-
-  if(GRPC_UPB_LIB)
-    # upb is used by recent gRPC versions
-    add_library(gRPC::upb UNKNOWN IMPORTED)
-    set_target_properties(gRPC::upb
-                          PROPERTIES IMPORTED_LOCATION "${GRPC_UPB_LIB}"
-                                     INTERFACE_INCLUDE_DIRECTORIES "${GRPC_INCLUDE_DIR}")
-    set(_GRPCPP_LINK_LIBRARIES "${_GRPCPP_LINK_LIBRARIES};gRPC::upb")
-  endif()
-
-  find_package(absl CONFIG)
-  if(absl_FOUND)
-    # Abseil libraries that recent gRPC versions depend on
-    set(_ABSL_LIBS
-        bad_optional_access
-        int128
-        raw_logging_internal
-        str_format_internal
-        strings
-        throw_delegate
-        time
-        time_zone)
-
-    foreach(_ABSL_LIB ${_ABSL_LIBS})
-      set(_GRPCPP_LINK_LIBRARIES "${_GRPCPP_LINK_LIBRARIES};absl::${_ABSL_LIB}")
-    endforeach()
-  endif()
-
   add_library(gRPC::grpc++ UNKNOWN IMPORTED)
   set_target_properties(gRPC::grpc++
                         PROPERTIES IMPORTED_LOCATION
-                                   "${GRPC_GRPCPP_LIB}"
-                                   INTERFACE_LINK_LIBRARIES
-                                   "${_GRPCPP_LINK_LIBRARIES}"
+                                   "${GRPCPP_IMPORTED_LOCATION}"
+                                   INTERFACE_COMPILE_OPTIONS
+                                   "${GRPCPP_COMPILE_OPTIONS}"
                                    INTERFACE_INCLUDE_DIRECTORIES
-                                   "${GRPC_INCLUDE_DIR}")
+                                   "${GRPCPP_INCLUDE_DIRECTORIES}"
+                                   INTERFACE_LINK_LIBRARIES
+                                   "${GRPCPP_LINK_LIBRARIES}"
```

Comment:
Could `GRPCPP_LINK_LIBRARIES` contain extraneous libraries that might cause problems here?

Reply:
If `grpc++.pc` is broken, it can be occurred. But it's a `grpc++.pc` problem. We should fix `grpc++.pc` instead of our `.cmake`.